### ISSUE TITLE
Support different line lengths

### ISSUE THREAD

**Issue by vaind** [dart-lang/pub-dev#3041]
It would be nice if you could support different line lengths, e.g. `-l 120` parameter to `pana`. While 80 is a recommended `dartfmt` default, some projects don't want to use that and instead opt for a different number. 

Currently, it's reported as a  "Health suggestion" on the package analysis tab:
#### Health suggestions
**Format `path/to/file.dart`.**
Run `dartfmt` to format `path/to/file.dart`.

---------------

I don't think it's a project health issue as long as the whole project is consistent with the formatting, albeit with a different line length than the default 80. While I understand these suggestions have no negative impact on the score I'd still prefer them not to show up on the page at all.

Is there something that can be done about it? Maybe some simple configuration file or a config in the pub.dev admin section?

**Comment by jonasfj** [comment]
I don't think we can or should do anything in the short term. But this is solid input for the debate about what lints we should use to score packages.

**Comment by techwn** [comment]
> I don't think we can or should do anything in the short term. But this is solid input for the debate about what lints we should use to score packages.

I think this is not something hard to complete, I think it should be done as soon as possible. 
As I know source code of flutter not follow 80 character per line.
And this will provide great develop experience and code readability.

**Comment by alexmercerind** [comment]
Allow us to have some max-lines attribute in the pubspec.yaml... fetch that on your score calculator.

**Comment by sigurdm** [comment]
From dart 3.7 the dart formatter allows a line-length (page-width) to be specified in analysis_options.yaml

See https://dart.dev/tools/dart-format#configuring-formatter-page-width

Pana is explicitly preserving this option:
https://github.com/dart-lang/pana/blob/19a2d6c5a6f96673f536b602c8121e16afd525b9/lib/src/analysis_options.dart#L113

So this should be working by now!

**Comment by alexey-pelykh** [comment]
@sigurdm there a chance I'm missing something, yet I've added the `line_width` (https://github.com/alexey-pelykh/UVCCamera/blob/main/flutter/analysis_options.yaml) yet the analysis still wants to reformat everything https://pub.dev/packages/uvccamera/score

**Comment by alexey-pelykh** [comment]
@sigurdm I'm likely missing something yet I've added `page_width` https://github.com/alexey-pelykh/UVCCamera/blob/main/flutter/analysis_options.yaml and yet pub.dev wants to reformat everything https://pub.dev/packages/uvccamera/score

**Comment by isoos** [comment]
@alexey-pelykh Thanks for reporting this! I think your package's layout and content is good, so this seems to be a bug. We will investigate, [tracking issue here](https://github.com/dart-lang/pana/issues/1448).

**Comment by isoos** [comment]
@alexey-pelykh: The pana fix is now on pub.dev.
